src/components/core/DpDataTable/DpDataTable.vue

```diff
@@ -375,8 +375,20 @@ export default {
       const firstRow = this.tableEl.firstChild
       const tableHeaders = Array.prototype.slice.call(firstRow.childNodes)
       tableHeaders.forEach(tableHeader => {
-        const width = tableHeader.getBoundingClientRect().width
-        tableHeader.style.width = width + 'px'
+        /**
+         * The convert of the component DpTableHeader.vue from a functional component to a template indicates,
+         * that the DOM Element structure changed, so that the tableHeaders doesnt just contains HTML elements,
+         * it contains other code parts which are not HTML elements.
+         * So you get an error, that the method getBoundingClientRect is not a function,
+         * because it needs the same dom structure as before and the belonging HTML elements.
+         * So here needs to be a check if the tableHeader.nodeName is a 'th' element for this case.
+         * A general check if the tableHeader.nodeName is an HTML element here is inconvenient,
+         * so here its okay to do the check with directly with the 'th' element.
+         */
+        if(tableHeader.nodeName === "TH") {
```

Comment:
```suggestion
        /**
         * Some of the child nodes of the first table row are not Element nodes but comments or text. 
         * This originates in the Vue template compiler leaving empty html comments when rendering
         * falsy `v-if` blocks. We allow only nodeType "Element" to access its `getBoundingClientRect` api.
         */
        if (tableHeader.nodeType === 1) {
```

Reply:
thanks for the advice. this solution with the check of the nodeType looks better than my solution. done https://github.com/demos-europe/demosplan-ui/pull/205/commits/9c36b6e1589b293dbd4b256ffa0880f977a8b045